apps/emqx/src/emqx_flapping.erl

```diff
@@ -82,24 +93,13 @@ detect(#{clientid := ClientId, peerhost := PeerHost, zone := Zone}) ->
                 [] ->
                     false
             end
-    end.
+    end;
+detect(_ClientId, _PeerHost, #{enable := false}) ->
+    false.
 
-get_policy(Keys, Zone) when is_list(Keys) ->
-    RootKey = flapping_detect,
-    Conf = emqx_config:get_zone_conf(Zone, [RootKey]),
-    lists:foldl(
-        fun(Key, Acc) ->
-            case maps:find(Key, Conf) of
-                {ok, V} -> Acc#{Key => V};
-                error -> Acc#{Key => emqx_config:get([RootKey, Key])}
-            end
-        end,
-        #{},
-        Keys
-    );
-get_policy(Key, Zone) ->
-    #{Key := Conf} = get_policy([Key], Zone),
-    Conf.
+%% with default, if we delete Zone at running time. we should not crash.
+get_policy(Zone) ->
+    emqx_config:get_zone_conf(Zone, [flapping_detect], ?DEFAULT_POLICY).
```

Comment:
I think even user delete the`default` zone, the `default` zone should still present in run time config
 thus we don't need to provide default ?DEFAULT_POLICY ?

Reply:
there is currently no fallback to `default` zone, right ?
so if `Zone` is deleted, providing a default here indeed can prevent crashing, but it just hides the missing zone reference issue to later.

Reply:
right.
1. `default` zone must be present in the runtime after the schema is loaded.
2. right, fallback to default here just hides the issue. Without fallback, we could catch some error in the CI if there is unexpected behavior.